client/src/Views/MessagesListInsightsChannel.tsx

```diff
@@ -1,18 +1,21 @@
 import { faEnvelope, faEnvelopeOpen } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { formatFullDate, isMessageReaded } from 'src/Helpers'
+import { MessageFeedback } from 'src/custom-types'
 import 'src/themes/App.scss'
 import classNames from 'classnames'
 import React from 'react'
 import Badge from 'react-bootstrap/Badge'
 import { usePlayerUiState } from '../Store/PlayerUi'
 
-const MessagesListChatChannel = ({ messages, markAllAsRead }) => {
+interface Props { messages: MessageFeedback[], markAllAsRead: () => void }
+
+const MessagesListChatChannel: React.FC<Props> = ({ messages, markAllAsRead }) => {
   const { currentWargame, selectedForce, selectedRole } = usePlayerUiState()
   const selectedForceId = selectedForce ? selectedForce.uniqid : ''
-  const isMessageRead = (message) => isMessageReaded(currentWargame, selectedForceId, selectedRole, message._id)
+  const isMessageRead = (message: MessageFeedback) => isMessageReaded(currentWargame, selectedForceId, selectedRole, message._id)
   const isMessageUnreadExists = messages.some(message => !isMessageRead(message))
-
+  
```

Comment:
Hello @lilitkarapetyan - I want to drop the human-readable `force` and `role` from the `messageDetailsFrom` block.  This is because they can be determined from the `forceId` and `roleId` fields.   It will also mean that if a role-name changes during a wargame, the `from` for old messages will use the new name, rather than the one at the time when the message was sent.

So, this class will have to collate a list of wargames and roles (a dictionary, I guess) and lookup the names of the force and role in order to display them.

Reply:
I wanted to seek clarification regarding the proposed changes to the Message Details From interface. Below are the modifications that have been suggested:  ```
export interface MessageDetailsFrom {
  readonly forceId?: ForceData['uniqid'];
  readonly forceColor: ForceData['color'];
  readonly roleId: Role['roleId'];
  iconURL: ForceData['iconURL'];
  name?: string;
}
```  Before proceeding with these changes, could you please confirm if my understanding aligns with the intended task?

Reply:
Have captured this issue here: https://github.com/serge-web/serge/issues/2813